src/status_im/contexts/wallet/send/input_amount/view.cljs

```diff
@@ -30,28 +30,54 @@
 (defn valid-input?
   [current v]
   (let [max-length          12
-        length-owerflow?    (>= (count current) max-length)
+        length-overflow?    (>= (count current) max-length)
         extra-dot?          (and (= v dot) (string/includes? current dot))
         extra-leading-zero? (and (= current "0") (= "0" (str v)))
         non-numeric?        (re-find not-digits-or-dot-pattern (str v))]
-    (not (or non-numeric? extra-dot? extra-leading-zero? length-owerflow?))))
+    (not (or non-numeric? extra-dot? extra-leading-zero? length-overflow?))))
+
+(defn- add-char-to-string
+  [s c idx]
+  (let [size (count s)]
+    (if (= size idx)
+      (str s c)
+      (str (subs s 0 idx)
+           c
+           (subs s idx size)))))
+
+(defn- move-input-cursor
+  ([input-selection-atom new-idx]
+   (move-input-cursor input-selection-atom new-idx new-idx))
+  ([input-selection-atom new-start-idx new-end-idx]
+   (let [start-idx (if (< new-start-idx 0) 0 new-start-idx)
+         end-idx   (if (< new-end-idx 0) 0 new-start-idx)]
+     (swap! input-selection-atom assoc :start start-idx :end end-idx))))
 
 (defn- normalize-input
-  [current v]
-  (cond
-    (and (string/blank? current) (= v dot))
-    (str "0" v)
+  [current v input-selection-atom]
+  (let [{:keys [start end]} @input-selection-atom]
+    (if (= start end)
```

Comment:
`(= start end)`

I'm only handling the cases when the cursor is not selecting text, validating everything seems really exhaustive.